agent/tool-scripts/pprof

```diff
@@ -101,88 +102,75 @@ done
-                    else
-                        echo "OPENSHIFT_PROFILE=web" >> $ose_openshift_master_api
-                        echo "OPENSHIFT_PROFILE=web" >> $ose_openshift_master_controllers
-                        echo "OPENSHIFT_PROFILE_PORT=6061" >> $ose_openshift_master_api
-                        systemctl restart atomic-openshift-master-api.service atomic-openshift-master-controllers.service
-                    fi
-                else
-                    if grep -q "^OPENSHIFT_PROFILE=web" $ose_openshift_master; then
-                        systemctl restart atomic-openshift-master
-                    else
-                        echo "OPENSHIFT_PROFILE=web" >> $ose_openshift_master
-                        systemctl restart atomic-openshift-master.service
-                    fi
-                fi
-            fi 
-            ;;
-        node)
-            if [ -e $origin-node ]; then 
-                if grep -q "^OPENSHIFT_PROFILE=web" $origin_node; then
-                    systemctl restart openshift-node.service
-                else
-                    echo "OPENSHIFT_PROFILE=web" >> $origin_node
-                    systemctl restart openshift-node.service
-                fi 
-            else
-                if grep -q "^OPENSHIFT_PROFILE=web" $ose_openshift_node; then
-                    systemctl restart atomic-openshift-node.service
-                else
-                    echo "OPENSHIFT_PROFILE=web" >> $ose_openshift_node
-                    systemctl restart atomic-openshift-node.service
-                fi
-            fi 
-    esac
+get_masters() {
+    declare -A master_pids
+    while true; do
+        i=1
+        for master in "${masters[@]}"; do
+            oc get --raw /debug/pprof/profile?seconds=${interval} --server https://${master}:8443 > ${master}_api_profile && ${tool_bin} tool ${tool} -text -seconds=${interval} -output $PPROF_TMPDIR/${master}/$(date +%Y-%m-%d:%H:%M:%S)_api_profile.txt /tmp/${master}_api_profile &
+            master_pids[${master}_api]=$!
+            oc get --raw /debug/pprof/heap --server https://${master}:8443 > ${master}_api_heap && ${tool_bin} tool ${tool} -text -seconds=${interval} -output $PPROF_TMPDIR/${master}/$(date +%Y-%m-%d:%H:%M:%S)_api_heap.txt /tmp/${master}_api_heap &
+            oc get --raw /debug/pprof/profile?seconds=${interval} --server https://${master}:8444 > ${master}_controller_profile && ${tool_bin} tool ${tool} -text -seconds=${interval} -output $PPROF_TMPDIR/${master}/$(date +%Y-%m-%d:%H:%M:%S)_controller_profile.txt /tmp/${master}_controller_profile &
+            master_pids[${master}_controller]=$!
+            oc get --raw /debug/pprof/heap --server https://${master}:8444 > ${master}_controller_heap && ${tool_bin} tool ${tool} -text -seconds=${interval} -output $PPROF_TMPDIR/${master}/$(date +%Y-%m-%d:%H:%M:%S)_controller_heap.txt /tmp/${master}_controller_heap &
```

Comment:
Why isn't this pid being tracked in `master_pids`?

Reply:
Heap collection is non-blocking

Reply:
What does that mean?  That the pid does not say around?  Should we then remove the trailing `&` as a result?

Reply:
When we do a profiling request there is a time period `${interval}` that the pid stays alive for, where the heap will just dump and return the heap. Given that we want all the heaps & profiles to be sampled simultaneously I don't think we want to prevent it from forking.

Reply:
Then it seems we need to also `wait` for these pids as well.

Reply:
Why do we need to wait for the heap pids?
We're kicking off a heap dump pid and a profile pid, the heap dump pid will never finish after the profile pid.

Reply:
Because it would seem that we don't want to allow an error condition to create pids without reaping them, filling a system. Making sure we wait for all pids we fork prevents that from happening.